compiler/rustc_passes/src/check_attr.rs

```diff
@@ -358,6 +358,17 @@ impl CheckAttrVisitor<'tcx> {
                 .emit();
             return false;
         }
+        let item_name = self.tcx.hir().name(hir_id);
+        if item_name.to_string() == doc_alias {
+            self.tcx
+                .sess
+                .struct_span_err(
+                    meta.span(),
+                    &format!("`#[doc(alias = \"...\")]` is the same as the item's name"),
+                )
+                .emit();
+            return false;
+        }
```

Comment:
Does this need to be a hard error? I think it would be better as just a warning. Unfortunately that probably means it has to wait for https://github.com/rust-lang/rust/pull/80527 so the lint level can be changed.

Reply:
Actually I just realized everything here is a hard error. So we should change all those to be lints at some point, but it doesn't need to block this PR.